plenum/server/node.py

```diff
@@ -2271,21 +2272,27 @@ def allLedgersCaughtUp(self):
         if self.num_txns_caught_up_in_last_catchup() == 0:
             self.catchup_rounds_without_txns += 1
         last_caught_up_3PC = self.ledgerManager.last_caught_up_3PC
-        if compare_3PC_keys(self.master_last_ordered_3PC,
-                            last_caught_up_3PC) > 0:
-            for replica in self.replicas.values():
-                if replica.isMaster:
-                    replica.caught_up_till_3pc(last_caught_up_3PC)
-                else:
-                    replica.catchup_clear_for_backup()
-            logger.info('{}{} caught up till {}'
-                        .format(CATCH_UP_PREFIX, self, last_caught_up_3PC),
-                        extra={'cli': True})
+        self.mode = Mode.synced
+
+        need_up_and_clear = compare_3PC_keys(self.master_last_ordered_3PC,
```

Comment:
Why can't we do this check inside `on_catch_up_finished`?

Reply:
Because in this case we firstly update master replica and can't update backups after, master_last_ordered_3PC will already be changed.

Reply:
We can revert replicas list for this "for" but it looks as an unobvious way.

Reply:
well, but we can just pass `master_last_ordered_3PC` then

Reply:
yes... thank you. Will fix it.